src/cdk/schematics/testing/resolve-bazel.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as path from 'path';
+
+/**
+ * Resolves a given path through Bazel while respecting the specified parent directory.
+ * Usually `require.resolve` could just be used for resolving runfiles relatively, but support
+ * for this has been removed and absolute manifest paths or the runfile helpers are now needed
+ * on Windows. This helper provides a resolution method that works similar on all platforms.
+ */
+export function resolveBazel(parent: string, relativePath: string) {
+  if (process.env['RUNFILES_MANIFEST_ONLY'] !== '1') {
+    return path.join(parent, relativePath);
+  }
+  // Imports the runfile helpers from the Bazel NodeJS rules. This should be ideally
+  // a separate package with types. https://github.com/bazelbuild/rules_nodejs/issues/1793.
+  // Note: We don't want to import this outside of this function as the runfile helpers are
+  // quite large we don't want to load them for every import to `@angular/cdk/testing/private`.
```

Comment:
I'm starting to feel like we should keep a centralized list of all the bazel-related rough edges / workarounds we have in place.

Reply:
Yeah, agreed. Maybe we should move this stuff into `tools/bazel` so that we can have a clear picture of what is specific to Bazel. We can do that as a follow-up if you think that makes sense.

Reply:
Up to you; you're the one who ends up dealing with it the most, so whatever makes your life easier